addons/calendar/models/calendar.py

```diff
@@ -0,0 +1,1014 @@
+                if len(str(one_ids).split('-')) == 1:
+                    ids = [int(one_ids)]
+                else:
+                    ids = [one_ids]
+        new_event = False
+
+        # Special write of complex IDS
+        for event_id in list(ids):
+            if len(str(event_id).split('-')) == 1:
+                continue
+
+            ids.remove(event_id)
+            real_event_id = calendar_id2real_id(event_id)
+
+            # if we are setting the recurrency flag to False or if we are only changing fields that
+            # should be only updated on the real ID and not on the virtual (like message_follower_ids):
+            # then set real ids to be updated.
+            if not values.get('recurrency', True) or not _only_changes_to_apply_on_real_ids(values.keys()):
+                ids.append(real_event_id)
+                continue
+            else:
+                event = self.browse(event_id)
+                if event.rrule:
+                    new_event = event._detach_one_event(values)
+
+        records = self.browse([int(event_id) for event_id in ids])
+        res = super(CalendarEvent, records).write(values)
+
+        # set end_date for calendar searching
+        if values.get('recurrency', True) and values.get('end_type', 'count') in ('count', unicode('count')) and \
+                (values.get('rrule_type') or values.get('count') or values.get('start') or values.get('stop')):
+            for record in records:
+                final_date = record._get_recurrency_end_date()
+                super(CalendarEvent, record).write({'final_date': final_date})
+
+        attendees_create = False
+        if values.get('partner_ids'):
+            attendees_create = records.create_attendees()
+
+        if (values.get('start_date') or values.get('start_datetime')) and values.get('active', True):
+            the_id = new_event or (ids and int(ids[0]))
+            if the_id:
+                if attendees_create:
+                    attendees_create = attendees_create[the_id]
+                    mail_to_ids = attendees_create['old_attendees'] - attendees_create['removed_attendees']
+                else:
+                    mail_to_ids = self.browse(the_id).attendee_ids
+
+                if mail_to_ids:
+                    mail_to_ids._send_mail_to_attendees(template_xmlid='calendar_template_meeting_changedate', email_from=self.env.user.email)
+        return bool(res)
+
+    @api.model
+    def create(self, vals):
+        self._set_date(vals)
+        res = super(CalendarEvent, self).create(vals)
+        final_date = res._get_recurrency_end_date()
+        res.write({'final_date': final_date})
+        res.create_attendees()
+        return res
+
+    def export_data(self, *args, **kwargs):
+        """ Override to convert virtual ids to ids """
+        return super(CalendarEvent, self.browse(list(set(get_real_ids(self.ids))))).export_data(*args, **kwargs)
```

Comment:
self.browse(set(get_real_ids(self.ids))) should be enough, _normalize_ids will do the job